src/ray/raylet/node_manager.cc

```diff
@@ -218,20 +218,19 @@ ray::Status NodeManager::RegisterGcs() {
     RAY_CHECK_OK(status);
     // Subscribe to resource changes.
     const auto &resources_changed =
-        [this](const ClientID &id,
-               const gcs::ResourceChangeNotification &resource_notification) {
-          if (resource_notification.IsAdded()) {
+        [this](const rpc::NodeResourceChange &resource_notification) {
+          auto id = ClientID::FromBinary(resource_notification.node_id());
+          if (0 == resource_notification.deleted_resources_size()) {
```

Comment:
Nit, better not use `if-else`, because one notification could carry both deleted resources and updated resources in the future.

Reply:
fixed